src/senaite/core/schema/gpscoordinatesfield.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of SENAITE.CORE.
+#
+# SENAITE.CORE is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright 2018-2024 by it's authors.
+# Some rights reserved, see README and LICENSE.
+
+import copy
+
+from bika.lims import api
+from bika.lims import senaiteMessageFactory as _
+from senaite.core.i18n import translate as t
+from senaite.core.schema.fields import BaseField
+from senaite.core.schema.interfaces import IGPSCoordinatesField
+from zope.interface import implementer
+from zope.schema import Dict
+
+SUBFIELDS = ("latitude", "longitude")
+
+# One meter resolution ~ 5 decimal places
+# ~11 centimeter resolution ~ 6 decimal places
+# One centimeter resolution ~ 7 decimal places
+# One millimeter resolution ~ 8 decimal places
+# https://xkcd.com/2170/
+DEFAULT_PRECISION = 7
+
+
+@implementer(IGPSCoordinatesField)
+class GPSCoordinatesField(Dict, BaseField):
+    """A field that handles GPS coordinates
+    """
+
+    def __init__(self, **kwargs):
+        default = kwargs.get("default")
+        kwargs["default"] = default or dict.fromkeys(SUBFIELDS, "")
+        super(GPSCoordinatesField, self).__init__(**kwargs)
+        self.precision = kwargs.get("precision") or DEFAULT_PRECISION
+
+    def set(self, object, value):
+        """Set the GPS coordinates field value
+        """
+        # ensure the value is a valid dict
+        coordinates = copy.deepcopy(self.default)
+        if value:
+            coordinates.update(value)
+
+        # convert values to strings with the right precision
+        template = "{:.%df}" % self.precision
```

Comment:
Would it make sense to do a type checking here if the precision was passed in?

Reply:
Addressed with https://github.com/senaite/senaite.core/pull/2565/commits/11bbbe811c5f500b0486285f52f6bd998be75ec3